### PR TITLE
Implementations for as_offset

### PR DESCRIPTION
**Pull request by nfarabullini**
Implementation of 
```python
 @field_operator(backend=fieldview_backend)
  def offset_index_field_fo(
      a: Field[[IDim, KDim], float64],
      offset_field: Field[[IDim, KDim], int64],
  ) -> Field[[IDim, KDim], float64]:
      return a(as_offset(Ioff, offset_field))
```
where `as_offset` shifts a field of values wrt to a field of offset values

### CODE REVIEW

File: tests/next_tests/ffront_tests/test_execution.py [hunk 1]
```diff
@@ -333,6 +332,40 @@ def astype_fieldop_float(b: Field[[IDim], int64]) -> Field[[IDim], int32]:
...
+
+    @field_operator(backend=fieldview_backend)
+    def dusk_index_fo(
+        a: Field[[IDim, KDim], float64],
+        offset_field: Field[[IDim, KDim], int64],
+        offset_i: Field[[IDim], int64],
+    ) -> Field[[IDim, KDim], float64]:
+        a_0 = a(as_offset(IDim, offset_i))
+        a_i = a_0(as_offset(IDim, offset_field))
+        a_i_k = a_i(as_offset(KDim, offset_field))
```
Comment (havogt) [inline tests/next_tests/ffront_tests/test_execution.py:353]: I only realized now that I was not precise in the description, it should be offsets not dimensions here, sorry for that (it's just that these offsets are then mapped to dimensions in the offset provider, that's why I made this mistake in the description)
```suggestion
        a_0 = a(as_offset(IOff, offset_i))
        a_i = a_0(as_offset(IOff, offset_field))
        a_i_k = a_i(as_offset(KOff, offset_field))
```

File: tests/next_tests/ffront_tests/test_execution.py [hunk 2]
```diff
@@ -333,6 +332,46 @@ def astype_fieldop_float(b: Field[[IDim], int64]) -> Field[[IDim], int32]:
...
+    a_I_float = np_as_located_field(IDim, KDim)(a_I_arr)
+    offset_field_arr = np.asarray(np.ones((size, size)), dtype=int64)
+    offset_i_arr = np.asarray(np.ones((size,)), dtype=int64)
+    offset_field = np_as_located_field(IDim, KDim)(offset_field_arr)
+    offset_i = np_as_located_field(IDim)(offset_i_arr)
+    out_I_float = np_as_located_field(IDim, KDim)(np.zeros((size, size), dtype=float64))
+    out_I_float_1 = np_as_located_field(IDim, KDim)(np.zeros((size, size), dtype=float64))
+
+    @field_operator(backend=fieldview_backend)
+    def dusk_index_fo(
```
Comment (havogt) [inline tests/next_tests/ffront_tests/test_execution.py:346]: remove dusk from this name

File: tests/next_tests/ffront_tests/test_execution.py [hunk 3]
```diff
@@ -333,6 +332,46 @@ def astype_fieldop_float(b: Field[[IDim], int64]) -> Field[[IDim], int32]:
     assert np.allclose(c_int32.array(), out_int_32)
 
 
+def test_dusk_indexfield(fieldview_backend):
```
Comment (havogt) [inline tests/next_tests/ffront_tests/test_execution.py:335]: call it something with offsetfield

**Review by DropD — CHANGES_REQUESTED** [review]
Just a small clarity improvement (probably copy pasta)

File: src/gt4py/next/ffront/experimental.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+@dataclass
+class BuiltInFunction:
+    __gt_type: ts.FunctionType
+
+    def __call__(self, *args, **kwargs):
+        """Act as an empty place holder for the built in function."""
+
+    def __gt_type__(self):
+        return self.__gt_type
```
Comment (havogt) [inline src/gt4py/next/ffront/experimental.py:28]: Should this be in a `ffront/common.py` or something?

Comment (nfarabullini) [reply src/gt4py/next/ffront/experimental.py:28]: if I move it to `common`I get this error:
```../../../src/gt4py/next/type_system/type_specifications.py:64: in DimensionType
    dim: func_common.Dimension
E   AttributeError: partially initialized module 'gt4py.next.common' has no attribute 'Dimension' (most likely due to a circular import)
```

**Review by DropD — APPROVED** [review]

**Pull request merged**